lightning/src/ln/msgs.rs

```diff
@@ -420,48 +421,51 @@ pub struct Stfu {
 	pub initiator: u8,
 }
 
-/// A splice message to be sent by or received from the stfu initiator (splice initiator).
-// TODO(splicing): Add spec link for `splice`; still in draft, using from https://github.com/lightning/bolts/pull/863
+/// A `splice_init` message to be sent by or received from the stfu initiator (splice initiator).
+///
+// TODO(splicing): Add spec link for `splice_init`; still in draft, using from https://github.com/lightning/bolts/pull/1160
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct Splice {
+pub struct SpliceInit {
 	/// The channel ID where splicing is intended
 	pub channel_id: ChannelId,
-	/// The genesis hash of the blockchain where the channel is intended to be spliced
-	pub chain_hash: ChainHash,
-	/// The intended change in channel capacity: the amount to be added (positive value)
-	/// or removed (negative value) by the sender (splice initiator) by splicing into/from the channel.
-	pub relative_satoshis: i64,
+	/// The amount the splice initiator is intending to add to its channel balance (splice-in)
+	/// or remove from its channel balance (splice-out).
+	pub funding_contribution_satoshis: i64,
 	/// The feerate for the new funding transaction, set by the splice initiator
 	pub funding_feerate_perkw: u32,
 	/// The locktime for the new funding transaction
 	pub locktime: u32,
 	/// The key of the sender (splice initiator) controlling the new funding transaction
 	pub funding_pubkey: PublicKey,
+	/// If set, only confirmed inputs added (by the splice acceptor) will be accepted
+	pub require_confirmed_inputs: Option<bool>,
```

Comment:
Doesn't look like the spec serializes any data (i.e., it's a zero-length TLV). So we may have to make this the unit type or define a type.

Reply:
Good catch. In fact, `require_confirmed_inputs` field was already present in OpenChannelV2, likewise with no data serialized.